diff --git a/src/nodes.rs b/src/nodes.rs
index 1ab2de5..96c2a64 100644
--- a/src/nodes.rs
+++ b/src/nodes.rs
@@ -31,6 +31,10 @@ macro_rules! node_matches {
     test,
     strum_discriminants(vis(pub(crate)), derive(strum::VariantArray, Hash))
 )]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum NodeValue {
     /// The root of every CommonMark document.  Contains **blocks**.
     Document,
@@ -253,6 +257,10 @@ pub enum NodeValue {
 
 /// Alignment of a single table cell.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum TableAlignment {
     /// Cell content is unaligned.
     None,
@@ -280,6 +288,10 @@ impl TableAlignment {
 
 /// The metadata of a table
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeTable {
     /// The table alignments
     pub alignments: Vec<TableAlignment>,
@@ -296,6 +308,10 @@ pub struct NodeTable {
 
 /// An inline [code span](https://github.github.com/gfm/#code-spans).
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeCode {
     /// The number of backticks
     pub num_backticks: usize,
@@ -309,6 +325,10 @@ pub struct NodeCode {
 
 /// The details of a link's destination, or an image's source.
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeLink {
     /// The URL for the link destination or image source.
     pub url: String,
@@ -322,6 +342,10 @@ pub struct NodeLink {
 
 /// The details of a wikilink's destination.
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeWikiLink {
     /// The URL for the link destination.
     pub url: String,
@@ -329,6 +353,10 @@ pub struct NodeWikiLink {
 
 /// The metadata of a list; the kind of list, the delimiter used and so on.
 #[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeList {
     /// The kind of list (bullet (unordered) or ordered).
     pub list_type: ListType,
@@ -358,6 +386,10 @@ pub struct NodeList {
 
 /// The metadata of a description list
 #[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeDescriptionItem {
     /// Number of spaces before the list marker.
     pub marker_offset: usize,
@@ -372,6 +404,10 @@ pub struct NodeDescriptionItem {
 
 /// The type of list.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum ListType {
     /// A bullet list, i.e. an unordered list.
     #[default]
@@ -383,6 +419,10 @@ pub enum ListType {
 
 /// The delimiter for ordered lists, i.e. the character which appears after each number.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum ListDelimType {
     /// A period character `.`.
     #[default]
@@ -403,6 +443,10 @@ impl ListDelimType {
 
 /// The metadata and data of a code block (fenced or indented).
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeCodeBlock {
     /// Whether the code block is fenced.
     pub fenced: bool,
@@ -432,6 +476,10 @@ pub struct NodeCodeBlock {
 
 /// The metadata of a heading.
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeHeading {
     /// The level of the header; from 1 to 6 for ATX headings, 1 or 2 for setext headings.
     pub level: u8,
@@ -446,6 +494,10 @@ pub struct NodeHeading {
 
 /// The metadata of an included HTML block.
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeHtmlBlock {
     /// The HTML block's type
     pub block_type: u8,
@@ -457,6 +509,10 @@ pub struct NodeHtmlBlock {
 
 /// The metadata of a footnote definition.
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeFootnoteDefinition {
     /// The name of the footnote.
     pub name: String,
@@ -467,6 +523,10 @@ pub struct NodeFootnoteDefinition {
 
 /// The metadata of a footnote reference.
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeFootnoteReference {
     /// The name of the footnote.
     pub name: String,
@@ -483,6 +543,10 @@ pub struct NodeFootnoteReference {
 
 /// The metadata of a multiline blockquote.
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeMultilineBlockQuote {
     /// The length of the fence.
     pub fence_length: usize,
@@ -493,6 +557,10 @@ pub struct NodeMultilineBlockQuote {
 
 /// An inline math span
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeMath {
     /// Whether this is dollar math (`$` or `$$`).
     /// `false` indicates it is code math
@@ -510,6 +578,10 @@ pub struct NodeMath {
 
 /// The metadata of an Alert node.
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeAlert {
     /// Type of alert
     pub alert_type: AlertType,
@@ -529,6 +601,10 @@ pub struct NodeAlert {
 
 /// The type of alert.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum AlertType {
     /// Useful information that users should know, even when skimming content
     #[default]
@@ -708,6 +784,10 @@ impl NodeValue {
 /// The struct contains metadata about the node's position in the original document, and the core
 /// enum, `NodeValue`.
 #[derive(Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct Ast {
     /// The node value itself.
     pub value: NodeValue,
@@ -745,6 +825,10 @@ const AST_NODE_SIZE_ASSERTION: [u8; 176] = [0; std::mem::size_of::<AstNode<'_>>(
 
 /// Represents the position in the source Markdown this node was rendered from.
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct Sourcepos {
     /// The line and column of the first character of this node.
     pub start: LineColumn,
@@ -791,6 +875,10 @@ impl From<(usize, usize, usize, usize)> for Sourcepos {
 
 /// Represents the 1-based line and column positions of a given character.
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct LineColumn {
     /// The 1-based line number of the character.
     pub line: usize,
@@ -902,6 +990,7 @@ impl<'a> From<Ast> for AstNode<'a> {
 
 /// Validation errors produced by [arena_tree::Node::validate].
 #[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
 pub enum ValidationError<'a> {
     /// The type of a child node is not allowed in the parent node. This can happen when an inline
     /// node is found in a block container, a block is found in an inline node, etc.
