diff --git a/src/parser/phoenix_heex.rs b/src/parser/phoenix_heex.rs
index 047223e..3c0129e 100644
--- a/src/parser/phoenix_heex.rs
+++ b/src/parser/phoenix_heex.rs
@@ -1,5 +1,9 @@
 /// Represents the type of Phoenix HEEx node.
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub enum HeexNode {
     /// A directive like `<% %>` or `<%= %>`.
     Directive,
@@ -15,6 +19,10 @@ pub enum HeexNode {
 
 /// The metadata of a Phoenix HEEx block-level element.
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
 pub struct NodeHeexBlock {
     /// The literal contents of the Phoenix block element, including delimiters.
     pub literal: String,
